File: Python/stats.py
from collections import Counter

class StatisticsCalculator:
    def __init__(self, data):
        self.data = sorted(data)

    def calculate_mean(self):
        return sum(self.data) / len(self.data)

    def calculate_median(self):
        n = len(self.data)
        mid = n // 2
        if n % 2 == 0:
            return (self.data[mid - 1] + self.data[mid]) / 2
        else:
            return self.data[mid]

    def calculate_mode(self):
        count = Counter(self.data)
        max_freq = max(count.values())
        return [k for k, v in count.items() if v == max_freq]

if __name__ == "__main__":
    data = [4, 2, 2, 3, 5, 2, 3, 4]
    calc = StatisticsCalculator(data)

    print("Mean:", round(calc.calculate_mean(), 2))
    print("Median:", round(calc.calculate_median(), 2))
    print("Mode(s):", " ".join(str(m) for m in calc.calculate_mode()))

File: C/stats.c
#include <stdio.h>
#include <stdlib.h>

int compare(const void* a, const void* b) {
    return (*(int*)a - *(int*)b);
}

double calculate_mean(int arr[], int size) {
    int sum = 0;
    for (int i = 0; i < size; i++) sum += arr[i];
    return (double)sum / size;
}

double calculate_median(int arr[], int size) {
    qsort(arr, size, sizeof(int), compare);
    if (size % 2 == 0)
        return (arr[size/2 - 1] + arr[size/2]) / 2.0;
    else
        return arr[size/2];
}

void calculate_mode(int arr[], int size) {
    int max_count = 0, count = 1;
    qsort(arr, size, sizeof(int), compare);
    printf("Mode(s): ");
    for (int i = 1; i <= size; i++) {
        if (i < size && arr[i] == arr[i - 1])
            count++;
        else {
            if (count > max_count) {
                max_count = count;
            }
            count = 1;
        }
    }

    count = 1;
    for (int i = 1; i <= size; i++) {
        if (i < size && arr[i] == arr[i - 1])
            count++;
        else {
            if (count == max_count)
                printf("%d ", arr[i - 1]);
            count = 1;
        }
    }
    printf("\n");
}

int main() {
    int arr[] = {4, 2, 2, 3, 5, 2, 3, 4};
    int size = sizeof(arr)/sizeof(arr[0]);

    printf("Mean: %.2f\n", calculate_mean(arr, size));
    printf("Median: %.2f\n", calculate_median(arr, size));
    calculate_mode(arr, size);

    return 0;
}
